src/index.js

```diff
@@ -56,15 +67,23 @@ function testUniqueness( element, selector )
  */
 function getUniqueCombination( element, items, tag )
 {
-  const combinations = getCombinations( items );
-  const uniqCombinations = combinations.filter( testUniqueness.bind( this, element ) );
-  if( uniqCombinations.length ) return uniqCombinations[ 0 ];
+  let combinations = getCombinations( items ),
+      firstUnique = getFirstUnique( element, combinations );
+
+  if( Boolean( firstUnique ) )
+  {
+      return firstUnique;
+  }
 
   if( Boolean( tag ) )
   {
-      const combinations = items.map( item => tag + item );
-      const uniqCombinations = combinations.filter( testUniqueness.bind( this, element ) );
-      if( uniqCombinations.length ) return uniqCombinations[ 0 ];
+      combinations = combinations.map( combination => tag + combination );
+      firstUnique = getFirstUnique( element, combinations );
+
+      if( Boolean( firstUnique ) )
```

Comment:
No it's not because firstUnique is either the return value of the function inside `.find()` or `undefined`. So it's not always a boolean. I personally would code this way but I have adapted to your style.